cardano-node/src/Cardano/Node/Startup.hs

```diff
@@ -208,7 +210,18 @@ prepareNodeInfo ptcl (SomeConsensusProtocol whichP pForInfo) tc nodeStartTime =
   prepareNodeName =
     case tcNodeName tc of
       Just aName -> return aName
-      Nothing    -> pack <$> getHostName
+      Nothing -> do
+        -- The user didn't specify node's name in the configuration.
+        -- In this case we should form node's name as "host:port", where 'host' and 'port'
+        -- are taken from '--host-addr' and '--port' CLI-parameters correspondingly.
+        let SocketConfig hostIPv4 hostIPv6 port _ = ncSocketConfig nc
+        case (getLast hostIPv4, getLast hostIPv6, getLast port) of
+          (Just ipv4, _,         Just p)  -> return . pack $ show ipv4 <> ":" <> show p
+          (Nothing,   Just ipv6, Just p)  -> return . pack $ show ipv6 <> ":" <> show p
+          (Just ipv4, _,         Nothing) -> return . pack $ show ipv4
+          (Nothing,   Just ipv6, Nothing) -> return . pack $ show ipv6
+          (Nothing,   Nothing,   Just p)  -> do h <- getHostName; return . pack $ show h <> ":" <> show p
+          _ -> pack <$> getHostName -- Nothing is specified, so just take a raw host name.
```

Comment:
I think this will be a bit clearer:
```suggestion
        hostName <- case (show <$> hostIPv4) <> (show <$> hostIPv6) of
          Last (Just addr) -> return addr
          Last  Nothing    -> getHostName
        return . pack $ hostName <> maybe "" ((":" ++) . show) (getLast port) 
```

Reply:
IMHO, my version is easier to read... But if you think it's better - I'll commit it.

Reply:
To be frank, I also find @coot 's version clearer.

Reply:
There are indeed two parts:
* taking out the port part out side of `case` analysis reduces the number of cases by half
* using `Last` monoid reduces the remaining three cases to just two

Reply:
note: I switched the order of elements in `<>` (we use `Last` rather than `First`).